src/main/java/com/weather/assignment/service/WeatherService.java

```diff
@@ -0,0 +1,44 @@
+package com.weather.assignment.service;
+
+import com.weather.assignment.document.WeatherElements;
+import com.weather.assignment.repository.WeatherRepository;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import java.util.Date;
+import java.util.Optional;
+import static com.weather.assignment.constants.WeatherConstants.OPEN_WEATHER_URL;
+
+@Service
+public class WeatherService {
+
+    @Autowired
```

Comment:
- The fields should always be private unless it is absolutely necessary.
- Always use constructor injection.